plugins/pluginmanagers/lazy.nix

```diff
@@ -160,40 +163,47 @@ in
           plugin:
           let
             keyExists = keyToCheck: attrSet: lib.elem keyToCheck (lib.attrNames attrSet);
+            converted =
```

Comment:
This whole `let` block (`pluginToLua` etc) is making me think there's bigger issues with this plugin that need to be cleaned up. I think that is best tackled as a follow up PR to this one.

It seems the `plugins` option is essentially syntactic sugar around the `settings.sepc` option? While we wouldn't normally do that, maybe it does have some merit here, if implemented in a sane way...

If we do decide it makes sense here, maybe the `plugins.lz-n` would benefit from a similar wrapper, although IIRC we already have a `plugins` option there, which is a structured-config representation of a _separate_ function call to `setup`.

Reply:
I think it's fine to have a `plugins` option, but the `cfg.plugins` value should be assigned to `settings.spec` _without_ using a `setupOptions` let-variable.

e.g.
```nix
extraConfig = cfg: {
  plugins.lazy.settings.spec =
    let
      # omitted
    in
      packedPlugins;
}
```